herbert_core/classes/data_loaders/@rundata/rm_masked.m

```diff
@@ -1,19 +1,37 @@
-function [S_m,Err_m,det_m]=rm_masked(this)
+function [S_m,Err_m,det_m]=rm_masked(obj,ignore_nan,ignore_inf)
 % method removes failed (NaN or Inf) data from the data array and deletes
 % detectors, which provided such signal
 %
-if isempty(this.S)||isempty(this.ERR)||isempty(this.det_par)
+if isempty(obj.S)||isempty(obj.ERR)||isempty(obj.det_par)
     error('RUNDATA:rm_masked',' signal, error and detectors arrays have to be defined\n');
 end
-if any(size(this.S)~=size(this.ERR))||(size(this.S,2)~=numel(this.det_par.x2))
+if any(size(obj.S)~=size(obj.ERR))||(size(obj.S,2)~=numel(obj.det_par.x2))
     error('RUNDATA:rm_masked',' signal error and detectors arrays are not consistent\n');
 end
+if ~exist('ignore_nan','var')
+    ignore_nan = true;
+end
+if ~exist('ignore_inf','var')
+    ignore_inf = true;
+end
 
-index_masked = (isnan(this.S)|(isinf(this.S))); % masked pixels
+
+if ignore_nan && ignore_inf
+    index_masked = (isnan(obj.S)|(isinf(obj.S))); % masked pixels
+elseif ignore_nan
+    index_masked = (isnan(obj.S));
```

Comment:
Are the outer brackets around the operations in these three cases required? They make the code harder to read